src/sentry/replays/lib/new_query/conditions.py

```diff
@@ -211,21 +211,39 @@ class IPv4Scalar(GenericBase):
     """IPv4 scalar condition class."""
 
     @staticmethod
-    def visit_eq(expression: Expression, value: str) -> Condition:
+    def visit_eq(expression: Expression, value: str | None) -> Condition:
+        if value is None:
+            return Condition(Function("isNull", parameters=[expression]), Op.EQ, 1)
         return Condition(expression, Op.EQ, Function("toIPv4", parameters=[value]))
 
     @staticmethod
-    def visit_neq(expression: Expression, value: str) -> Condition:
+    def visit_neq(expression: Expression, value: str | None) -> Condition:
+        if value is None:
+            return Condition(Function("isNull", parameters=[expression]), Op.EQ, 0)
         return Condition(expression, Op.NEQ, Function("toIPv4", parameters=[value]))
 
     @staticmethod
-    def visit_in(expression: Expression, value: list[str]) -> Condition:
-        values = [Function("toIPv4", parameters=[v]) for v in value]
+    def visit_in(expression: Expression, value_list: list[str | None]) -> Condition:
+        values = [Function("toIPv4", parameters=[v]) for v in value_list if v is not None]
+        if None in value_list:
+            return Or(
```

Comment:
Not sure if it's ok to return `snuba_sdk.conditions.Or/And` types here, instead of `snuba_sdk.Condition`. Or if we should maybe add a type annotation.

Passes typecheck, but my IDE is complaining. Maybe mypy is skipping stuff for snuba_sdk. @cmanallen do you have any thoughts?

Same applies to the visit_in and visit_not_in in aggregate.py